internal/storage/storage_handle_test.go

```diff
@@ -20,36 +61,54 @@ func getDefaultStorageClientConfig() (clientConfig storageClientConfig) {
 	}
 }
 
-func invokeAndVerifyStorageHandle(t *testing.T, sc storageClientConfig) {
-	handleCreated, err := NewStorageHandle(context.Background(), sc)
+func (t *StorageHandleTest) TestBucketHandleWhenBucketExists() {
+	server, err := createFakeServer()
```

Comment:
should we create the server once for all tests and add a client variable to StorageHandleTest struct.
Any specific reason we need one server for every test?